enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,258 @@
+
+To enable this using BGP, a BGP speaker must run on each Node where we would
+like to serve as a router into the cluster network.  Each Node can publish a
+route for the Pod and Service network and upstream routers may use ECMP to
+distribute traffic among the Nodes.
+
+A future enhancement needs to be filed to pursue this integration in more
+detail.
+
+### IP Anycast
+
+In both the [External Service Load Balancing](#external-service-load-balancing)
+and the [Exposing Pods or Services
+Directly](#exposing-pods-or-services-directly) sections, we discuss advertising
+some IP addresses via BGP to make them accessible from outside of the cluster.
+An extension would be to allow advertising the same IP address from more than
+one cluster (IP Anycast).
+
+This is particularly applicable to edge use cases where a client would be
+routed to the instance of a particular service IP with the fewest hops
+possible. This allows for regional affinity as the client is routed to the
+logically closest Node. This also allows for horizontal scalability, as
+additional clusters and Nodes could be created to spread client load across
+multiple service instances. Mobile clients may be expected to roam between
+endpoints as the number of hops changes with location, so this is suited to
+either fully stateless services over UDP such as DNS or video streaming, or
+semi-stateless services over TCP such as RESTful API endpoints. This technique
+could be used to load balance pods or services directly, or to expose multiple
+LoadBalancer instances.
+
+### Virtual Network Interconnect and Hybrid Cloud
+
+The Pod network is by default an isolated virtual network.  There is sometimes
+a desire to interconnect multiple virtual networks together.  This could be
+between multiple Kubernetes clusters, but it doesn't have to be the case.
+
+BGP, specifically MP-BGP, is commonly used as a signaling protocol, together
+with other data planes technologies like MPLS and VXLAN to interconnect virtual
+network at L2 or L3.
+
+BGP EVPN (Ethernet VPN, or Virtual Private Network) is a BGP technology that is
+discussed in the context of this use case.  BGP EVPN can be used to advertise
+MAC addresses and IP-MAC address bindings.  BGP EVPN is not a full solution to
+this use case by itself, as it is a technology that can help implement part of
+the control plane.  There are more details that must be explored, including
+what the data plane looks like between the two clusters.
+
+BGP and MPLS is heavily used in the Telco area, and by the Cloud providers to
+interconnect their facilities to the Cloud, i.e. [AWS Direct
+connect](https://d1.awsstatic.com/whitepapers/Networking/integrating-aws-with-multiprotocol-label-switching.pdf).
+
+This is a significantly non-trivial effort.  Future enhancements are needed to
+discuss this in much more detail.
+
+## Common Design Considerations
+
+This section summarizes the common design considerations for OpenShift in
+support of BGP related features.
+
```

Comment:
```suggestion
### Administrator Experience

Establishing BGP sessions is not a simple task, and requires a number of unavoidable manual steps and site-specific parameters. It also requires coordination between (what is likely) multiple departments or groups within a typical organization. Politically, the ability to speak BGP to a network grants significant privileges, since (by default) BGP sessions are trusted completely. A misbehaving BGP participant can easily cause massive outages -- even to networks far outside the OpenShift environment. Any design must take this in to account.

Technically, any solution must make it easy for administrators to configure the same BGP peering parameters as they would expect on a router. In addition to the required settings (e.g. AS, peer, password), administrators expect additional knobs that make BGP usable (e.g. communities, timing parameters, multihop, AS prepends). The current status of all peerings must be exposed, since administrators rely on this to debug their networks and trust that OpenShift is not misbehaving.

Documentation and training should reflect the complexity of BGP-based operations, and potential users should be made aware of any prerequisites and technical challenges. Everyone should be aware of the buy-in required for enabling BGP. 
```

Reply:
Very nice addition!  Thanks!